packages/api/lambdas/internal-reconciliation-report.js

```diff
@@ -59,82 +60,90 @@ async function internalRecReportForCollections(recReportParams) {
   );
 
   const collectionPgModel = new CollectionPgModel();
-  const knex = await getKnexClient();
-
-  // get collections from database and sort them, since the scan result is not ordered
-  const [updatedAtRangeParams, dbSearchParams] = convertToDBCollectionSearchObject(recReportParams);
-
-  const dbCollectionsSearched = await collectionPgModel.searchWithUpdatedAtRange(
-    knex,
-    dbSearchParams,
-    updatedAtRangeParams
-  );
-
-  // TODO - improve this sort
-  const dbCollectionItems = sortBy(
-    filterDBCollections(dbCollectionsSearched, recReportParams),
-    ['name', 'version']
-  );
-
-  let okCount = 0;
-  const withConflicts = [];
-  let onlyInEs = [];
-  let onlyInDb = [];
-
-  const fieldsIgnored = ['timestamp', 'updatedAt', 'createdAt'];
-  let nextEsItem = await esCollectionsIterator.peek();
-  let nextDbItem = dbCollectionItems.length !== 0
-    ? translatePostgresCollectionToApiCollection(dbCollectionItems[0])
-    : undefined;
-
-  while (nextEsItem && nextDbItem) {
-    const esCollectionId = constructCollectionId(nextEsItem.name, nextEsItem.version);
-    const dbCollectionId = constructCollectionId(nextDbItem.name, nextDbItem.version);
+  const knex = recReportParams.knex || await getKnexClient();
+
+  try {
```

Comment:
Is it necessary to add try/catch here?  We can try/catch the error in upper caller function to simplify the code, and you have already added try/catch in createInternalReconciliationReport, and log error there.

Reply:
That's true, I can update

Reply:
https://github.com/nasa/cumulus/pull/3332/commits/ff7b1d42578019db98a58d7ebc1032203f7e853b#diff-d06392e59ba6c4315b7509efe5ae98c2ad79eab742333baf145d95d4d9e266a8R65